cmd/jujud/main.go

```diff
@@ -181,6 +181,7 @@ func jujuDMain(args []string, ctx *cmd.Context) (code int, err error) {
 	jujud.Register(unitAgent)
 
 	jujud.Register(NewUpgradeMongoCommand())
+	jujud.Register(agentcmd.NewCheckConnectionCommand(agentConf, agentcmd.ReallyConnect))
```

Comment:
So all we need here is the data dir from the current conf right?  Let's pass that in instead of the whole conf which we are just going to overwrite anyway.